python-package/lightgbm/basic.py

```diff
@@ -2999,6 +2998,9 @@ def update(self, train_set=None, fobj=None):
             if not self.__set_objective_to_none:
                 self.reset_parameter({"objective": "none"}).__set_objective_to_none = True
             grad, hess = fobj(self.__inner_predict(0), self.train_set)
+            if self.num_model_per_iteration() > 1:
```

Comment:
Is it safe to use `_Booster__num_class` here instead to avoid the lib call? I don't fully understand where `__num_class` gets converted to `__Booster__num_class`.

Reply:
I mean that the attribute changes name. I see it's used as `self.__num_class` in some places but if I add a breakpoint at that line it doesn't have that attribute but has `self._Booster__num_class` which is the part that confuses me. Do you think the performance impact of calling the lib on each iteration is noticeable and should be changed to use the attribute instead?